lib/core-classes/DeviceService.ts

```diff
@@ -310,6 +318,16 @@ export class DeviceService {
     });
     device._source.assetId = null;
 
+    if(Array.isArray(asset._source.deviceIds)) {
```

Comment:
You should modify the `BaseAsset` class to ensure that `deviceIds` is always an array as we do for `measures`